### DISCUSSION TITLE
Can we install this without cert-manager

### DISCUSSION THREAD

**Discussion by rverma-dev** [actions/actions-runner-controller#532] [Questions] [answered]
We already have our self-signed certificates or ou certifications. Is it possible to use them instead of cert manager?

**Comment by mumoshu** [comment, marked as answer]
@rverma-nsl Maybe. As long as you can create your own certificate that can be a replacement for https://github.com/actions-runner-controller/actions-runner-controller/blob/05fb8569b33555e2ecede2c59095405e22dc3005/charts/actions-runner-controller/templates/certificate.yaml#L12-L24, it should work.

**Comment by michaelst** [comment]
if we don't use the webhooks this should not be required right?

**Reply by mumoshu** [reply]
That's technically right. But allowing so/making the webhook optional can be another maintenance burden for me. I'm going to add features that leverages admission webhooks. And everyone, including me, can't be sure what can block admission webhooks on what platform. We need to reduce variables and have more information and documentation, rather than just making it optional.

Also see https://github.com/actions-runner-controller/actions-runner-controller/discussions/537

**Reply by mumoshu** [reply]
That said, it's free for you to fork our chart/manifest to rip off the admission webhook configs and use that. But please be sure to mention those customizations made by yourself, when you come back to report issues and ask for user support in the future.

**Comment by rverma-dev** [comment]
But there are other projects who uses admission webhook and leverage helm
to generate ssl certs

Few of them I have been using
1. Aws load balancer controler
2. Kyverno
3. Flux
4. Ebs CSI driver.

On Tue 11 May, 2021, 7:06 AM Yusuke Kuoka, ***@***.***> wrote:

> That said, it's free for you to fork our chart/manifest to rip off the
> admission webhook configs and use that. But please be sure to mention those
> customizations made by yourself, when you come back to report issues and
> ask for user support in the future.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/actions-runner-controller/actions-runner-controller/discussions/532#discussioncomment-721316>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATCTF5QKIEPZTQDM4N36UBTTNCCYHANCNFSM44RKBP7Q>
> .
>

-- 


============================================ CONFIDENTIALITY NOTICE 
============================================ 
The contents of this email 
message and any attachments are intended solely for the addressee(s) and 
may contain confidential and or privileged information and may be legally 
protected from disclosure.If you are not the intended recipient of this 
message or their agent, or if this message has been addressed to you in 
error, please immediately alert the sender by reply email and then delete 
this message and any attachments.If you are not the intended recipient, you 
are hereby notified that any use, dissemination, copying, or storage of 
this message or its attachments or taking any action in reliance on the 
contents of this information is strictly prohibited.The integrity and 
security of this message cannot be guaranteed on the internet.Organization 
or the sender, does not accept any liability for any damages caused by 
errors, omissions or virus transmission connected to the contents of this 
message.
=================================================================================================================

**Comment by billyshambrook** [comment]
@mumoshu would you be open to a PR to make cert-manager optional (enabled by default for backwards compatibility), similar to how aws-load-balancer-controller chart does it here? https://github.com/aws/eks-charts/pull/261/files

Thanks!
